js/src/block-editor/hooks/useMedia.js

```diff
@@ -58,6 +59,13 @@ const useMedia = ( fieldValue, onChange, allowedTypes ) => {
 		return select( 'core' ).getMedia( fieldValue );
 	} );
 
+	/* @type {function|undefined} */
+	const mediaUpload = useSelect( ( select ) => {
+		// @ts-ignore The function isn't in the declaration file.
+		const { getSettings } = select( blockEditorStore );
+		return getSettings()?.mediaUpload || legacyMediaUpload;
```

Comment:
`legacyMediaUpload` is needed because the GCB editor doesn't have the block editor settings with `mediaUpload()`